libgeopmd/src/LevelZero.cpp

```diff
@@ -32,7 +33,30 @@ namespace geopm
                             "variable is set, must use service to access LevelZero in this case.",
                             GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
         }
-
+        if (geopm::has_cap_sys_admin()) {
+            // LevelZeroIOGroup is only loaded when PID has CAP_SYS_ADMIN
+            bool do_warn = false;
+            if (getenv("ZE_ENABLE_SYSMAN") == nullptr ||
+                std::string(getenv("ZE_ENABLE_SYSMAN")) != "1") {
+                do_warn = true;
+                std::cerr << "Warning: <geopm> LevelZero features require the environment variable setting \"ZE_ENABLE_SYSMAN=1\".\n";
+            }
+            if (getenv("ZE_FLAT_DEVICE_HIERARCHY") == nullptr ||
+                std::string(getenv("ZE_FLAT_DEVICE_HIERARCHY")) != "COMPOSITE") {
+                if (do_warn == false) {
+                    std::cerr << "Warning: <geopm> ";
+                }
+                else {
+                    std::cerr << "                 ";
+                }
+                do_warn = true;
+                std::cerr << "LevelZero features require the environment variable setting \"ZE_FLAT_DEVICE_HIERARCHY=COMPOSITE\".\n";
+            }
+            if (do_warn) {
+                std::cerr << "                 This process may corrupt the topology cache file.  Run the following commands as root to fix:\n"
+                          << "                     rm -f /run/geopm/geopm-topo-cache; ZE_ENABLE_SYSMAN=1 ZE_FLAT_DEVICE_HIERARCHY=COMPOSITE geopmread --cache\n";
```

Comment:
Might be nice to only warn if we are actually creating a topo cache file, but plumbing that turns out to be a bit tricky.